ext/curl/interface.c

```diff
@@ -1631,12 +1631,19 @@ static bool php_curl_set_callable_handler(zend_fcall_info_cache *const handler_f
 }
 
 
-#define HANDLE_CURL_OPTION_CALLABLE_PHP_CURL_USER(curl_ptr, constant_no_function, handler_type) \
+#define HANDLE_CURL_OPTION_CALLABLE_PHP_CURL_USER(curl_ptr, constant_no_function, handler_type, default_method) \
 	case constant_no_function##FUNCTION: { \
 		bool result = php_curl_set_callable_handler(&curl_ptr->handlers.handler_type->fcc, zvalue, is_array_config, #constant_no_function "FUNCTION"); \
 		if (!result) { \
 			return FAILURE; \
 		} \
+		if (!curl_ptr->handlers.handler_type) { \
+			return SUCCESS; \
+		} \
+		if (!ZEND_FCC_INITIALIZED(curl_ptr->handlers.handler_type->fcc)) { \
+			curl_ptr->handlers.handler_type->method = default_method; \
+			return FAILURE; \
```

Comment:
This should return `SUCCESS`. Setting options to `null` is allowed from what I understand. If you return here, you'll stop updating the other options in `curl_setopt_array`.